### PR TITLE
Handle postalcodes as polygons

### LINKED ISSUE #72: Load postal codes as boundaries/polygons
# Problem statement

Postal codes are not single points, a city might have multiple postal codes and we have to load them as polygons.

# Proposed solution

Load postal codes as boundaries (polygon) like cities or regions.

### PR DESCRIPTION
**Pull request by pierDipi**
# New features and improvements

- Load postal codes as boundaries/polygons.

# Related issue

Closes #72

### CODE REVIEW

File: src/main/scala/it/agilelab/bigdata/gis/domain/managers/IndexSet.scala [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package it.agilelab.bigdata.gis.domain.managers
+
+import it.agilelab.gis.domain.models.{ OSMBoundary, OSMHouseNumber, OSMStreetAndHouseNumber }
+import it.agilelab.gis.domain.spatialList.GeometryList
+
+case class IndexSet(
```
Comment (andr3a87) [inline src/main/scala/it/agilelab/bigdata/gis/domain/managers/IndexSet.scala:6]: scaladoc

File: src/main/scala/it/agilelab/gis/domain/loader/ShapeFileReader.scala [hunk 2]
```diff
@@ -16,8 +16,16 @@ object ShapeFileReader {
 
   implicit class SimpleFeatureWrapper(ft: SimpleFeature) {
 
-    def geom[G <: jts.Geometry: Manifest]: Option[G] =
-      ft.getAttribute(0) match {
+    def geom[G <: jts.Geometry: Manifest]: Option[G] = geom(0)
```
Comment (andr3a87) [inline src/main/scala/it/agilelab/gis/domain/loader/ShapeFileReader.scala:19]: is used this method?

File: src/main/scala/it/agilelab/gis/domain/loader/ShapeFileReader.scala [hunk 3]
```diff
@@ -16,8 +16,16 @@ object ShapeFileReader {
...
+
+    def geom[G <: jts.Geometry: Manifest](idx: Int): Option[G] =
```
Comment (andr3a87) [inline src/main/scala/it/agilelab/gis/domain/loader/ShapeFileReader.scala:21]: scaladoc

File: src/main/scala/it/agilelab/gis/domain/loader/ShapeFileReader.scala [hunk 4]
```diff
@@ -16,8 +16,16 @@ object ShapeFileReader {
...
+      ft.getAttribute(idx) match {
+        case g: G => Some(g)
+        case _    => None
+      }
+
+    def geom[G <: jts.Geometry: Manifest](attribute: String): Option[G] =
```
Comment (andr3a87) [inline src/main/scala/it/agilelab/gis/domain/loader/ShapeFileReader.scala:27]: scaladoc

File: src/main/scala/it/agilelab/gis/core/utils/ManagerUtils.scala [hunk 5]
```diff
@@ -12,7 +12,7 @@ object ManagerUtils {
       boundary: BoundaryPathGroup,
       postalCodes: Array[Path],
       roads: Array[Path],
-      houseNumbers: Option[Path]
+      houseNumbers: Array[Path]
```
Comment (andr3a87) [inline src/main/scala/it/agilelab/gis/core/utils/ManagerUtils.scala:15]: why?

Comment (pierDipi) [reply src/main/scala/it/agilelab/gis/core/utils/ManagerUtils.scala:15]: House numbers might be distributed in multiple files for a given country, for example, we have them for each region.

**Review by andr3a87 — APPROVED** [review]

**Pull request merged**